specification/workloads/resource-manager/Microsoft.Workloads/connectors/preview/2023-10-01-preview/backupTypes.json

```diff
@@ -0,0 +1,849 @@
+    },
+    "InstantRPAdditionalDetails": {
+      "type": "object",
+      "description": "Instant recovery point additional details.",
+      "properties": {
+        "azureBackupRGNamePrefix": {
+          "description": "Azure backup resource group name prefix.",
+          "type": "string"
+        },
+        "azureBackupRGNameSuffix": {
+          "description": "Azure backup resource group name suffix.",
+          "type": "string"
+        }
+      }
+    },
+    "SchedulePolicy": {
+      "description": "Base class for backup schedule.",
+      "required": [
+        "schedulePolicyType"
+      ],
+      "type": "object",
+      "properties": {
+        "schedulePolicyType": {
+          "description": "This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.",
+          "type": "string"
+        }
+      },
+      "discriminator": "schedulePolicyType"
+    },
+    "RetentionPolicy": {
+      "description": "Base class for retention policy.",
+      "required": [
+        "retentionPolicyType"
+      ],
+      "type": "object",
+      "properties": {
+        "retentionPolicyType": {
+          "description": "This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.",
+          "type": "string"
+        }
+      },
+      "discriminator": "retentionPolicyType"
+    },
+    "AzureVmWorkloadProtectionPolicy": {
+      "description": "Azure VM (Mercury) workload-specific backup policy.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProtectionPolicy"
+        }
+      ],
+      "properties": {
+        "workLoadType": {
+          "description": "Type of workload for the backup management",
+          "enum": [
+            "Invalid",
+            "VM",
+            "FileFolder",
+            "AzureSqlDb",
+            "SQLDB",
+            "Exchange",
+            "Sharepoint",
+            "VMwareVM",
+            "SystemState",
+            "Client",
+            "GenericDataSource",
+            "SQLDataBase",
+            "AzureFileShare",
+            "SAPHanaDatabase",
+            "SAPAseDatabase",
+            "SAPHanaDBInstance"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "WorkloadType",
+            "modelAsString": true
+          }
+        },
+        "settings": {
+          "$ref": "#/definitions/Settings",
+          "description": "Common settings for the backup management"
+        },
+        "subProtectionPolicy": {
+          "description": "List of sub-protection policies which includes schedule and retention",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SubProtectionPolicy"
+          },
+          "x-ms-identifiers": []
+        },
+        "makePolicyConsistent": {
+          "description": "Fix the policy inconsistency",
+          "type": "boolean"
+        }
```

Comment:
Typical recommendation is to replace boolean with enum offering extensibility. However, not stressing on this since this has been reviewed in the previous private repo merge.